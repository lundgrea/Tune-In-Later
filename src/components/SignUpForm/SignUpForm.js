import React, { Component } from 'react';
import { postUser } from "../../apiCalls/apiCalls";


export class SignUpForm extends Component {
  constructor() {
    super();
      this.state = {
        name: '',
        email: '',
        password: ''
      }
  }

  handleInputChange = (e) => {
    const {name, value} = e.target
    this.setState({
      [name]: value
    })
  }

  createAccount = (e) => {
    e.preventDefault();
    postUser(this.state);
    this.setState({
      name: '',
      email: '',
      password: ''
    })
  }


  render(){
    return (
      <form onSubmit={this.createAccount}>
        <input 
        type='text' 
        value={this.state.name} 
        placeholder='Joe Don' 
        name='name' 
        onChange={this.handleInputChange} 
        ></input>
        <input 
        type='text' 
        value={this.state.email} 
        placeholder='example@example.com' 
        name='email' 
        onChange={this.handleInputChange} 
        ></input>
        <input 
        type='text' 
        value={this.state.password} 
        placeholder='password' 
        name='password' 
        onChange={this.handleInputChange} 
        ></input>
        <button>Create Account</button>
      </form>
    )
  }
}