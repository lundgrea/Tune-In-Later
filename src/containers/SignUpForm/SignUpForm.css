.signup-button {
  width: 380px;
  margin: 10px 10px 10px 10px;
  height: 40px;
  font-size: 16px;
  padding: 10px;
  border: none;
  border-radius: 8px;
}

.signup-input {
  width: 80%;
  font-size: 16px;
  padding: 10px;
  margin: 10px 10px 10px 10px;
  border: none;
  border-radius: 8px;
}

.signin-form {
  flex-direction: column;
  display: flex;
  align-items: center;
  width: 100%;
  min-height: 25vh;
  justify-content: space-around;
}

.signin-form-container {
  min-height: 30vh;
  width: 50%;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  background-color: #a8d0e6;
  padding: 20px 10px 30px 20px;
}
