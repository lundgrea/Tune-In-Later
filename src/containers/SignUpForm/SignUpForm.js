import React, { Component } from 'react';
import { postUser } from "../../apiCalls/apiCalls";
import { handleErrors } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import './SignUpForm.css'
 
export class SignUpForm extends Component {
  constructor(props) {
    super(props);
      this.state = {
        name: '',
        email: '',
        password: ''
      }
  }

  handleInputChange = (e) => {
    const {name, value} = e.target
    this.setState({
      [name]: value
    })
  }

  createAccount = async (e) => {
    e.preventDefault();
    const user = await postUser(this.state);
    if(user.error) {
      this.props.handleError('Email has already been used')
    } else {
      this.setState({
        name: '',
        email: '',
        password: ''
      })
    }
  }


  render(){
    return (
      <article className="signin-form-container">
      <form className="signin-form"onSubmit={this.createAccount}>
        <h3>Create New Account</h3>
        <input 
        type='text' 
        value={this.state.name} 
        placeholder='Joe Don' 
        name='name' 
        onChange={this.handleInputChange} 
        className="signup-input"
        ></input>
        <input 
        type='text' 
        value={this.state.email} 
        placeholder='example@example.com' 
        name='email' 
        onChange={this.handleInputChange} 
        className="signup-input"
        ></input>
        <input 
        type='password' 
        value={this.state.password} 
        placeholder='password' 
        name='password' 
        onChange={this.handleInputChange} 
        className="signup-input"
        ></input>
        <button className="signup-input">Create Account</button>
      </form>
      </article>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  handleError: error => dispatch(handleErrors(error))
});

 export default connect(null, mapDispatchToProps)(SignUpForm)


 SignUpForm.propTypes = {
   handleError: PropTypes.func.isRequired
 }