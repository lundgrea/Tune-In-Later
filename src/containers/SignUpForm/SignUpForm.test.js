import React from "react";
import { shallow } from "enzyme";
import { SignUpForm } from "./SignUpForm";

describe("SignUpForm", () => {
  let wrapper;
  const mockHandleError = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SignUpForm handleError={mockHandleError} />);
  });

  it("should match the snapshot with all of the correct information passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleInputChange method when change happens in the name input", () => {
    const mockEvent = { target: { name: "name", value: "yolo" } };
    wrapper
      .find("input")
      .at(0)
      .simulate("change", mockEvent);
    expect(wrapper.state("name")).toEqual("yolo");
  });

  it("should call handleInputChange method when change happens in the email input", () => {
    const mockEvent = { target: { name: "email", value: "yolo@yolo.com" } };
    wrapper
      .find("input")
      .at(1)
      .simulate("change", mockEvent);
    expect(wrapper.state("email")).toEqual("yolo@yolo.com");
  });

  it("should call handleInputChange method when change happens in the password input", () => {
    const mockEvent = { target: { name: "password", value: "password" } };
    wrapper
      .find("input")
      .at(2)
      .simulate("change", mockEvent);
    expect(wrapper.state("password")).toEqual("password");
  });
});
