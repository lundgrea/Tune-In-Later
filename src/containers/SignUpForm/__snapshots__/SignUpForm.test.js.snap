// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match the snapshot with all of the correct information passed in 1`] = `
<article
  className="signin-form-container"
>
  <form
    className="signin-form"
    onSubmit={[Function]}
  >
    <h3>
      Create New Account
    </h3>
    <input
      className="signup-input"
      name="name"
      onChange={[Function]}
      placeholder="Joe Don"
      type="text"
      value=""
    />
    <input
      className="signup-input"
      name="email"
      onChange={[Function]}
      placeholder="example@example.com"
      type="text"
      value=""
    />
    <input
      className="signup-input"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <button
      className="signup-input"
    >
      Create Account
    </button>
  </form>
</article>
`;
